packages/reactive-element/src/decorators/custom-element.ts

```diff
@@ -13,6 +13,125 @@
 
 import type {Constructor} from './base.js';
 
+declare global {
+  interface CustomElementRegistry {
+    getName: (ctor: CustomElementConstructor) => string | null;
+  }
+}
+
+export type DefineErrorCallback = (
+  error: Error,
+  tagName: string,
+  ctor: CustomElementConstructor
+) => void;
+
+/**
+ *
+ * @param tagName The tag name of the custom element to define.
+ * @param ctor The constructor of the custom element to define
+ * @param onError An optional function which can handle a define exception
+ * error. If provided, the exception is caught.
+ * @param onError
+ * @returns void
+ */
+export const define = (
+  tagName: string,
+  ctor: CustomElementConstructor,
+  onError?: DefineErrorCallback
```

Comment:
I would put the `onError` callback in an options object. And we might want to pass that options object to the callback. This would let users place metadata in the object, like version number:

```ts
import {customElement, warnOnIncompatibleVersions} from 'lit/decorators.js';

@customElement('x-foo', {onError: warnOnIncompatibleVersions, version: '1.0.0'}
export class XFoo extends LitElement {}
```

Authors would be responsible for keeping the version in sync, hopefully with a build tool.

We could vend a common sentinel value to use as the version so that common tools could be made to replace it:

```ts
import {customElement, warnOnIncompatibleVersions, packageJsonVersion} from 'lit/decorators.js';

@customElement('x-foo', {onError: warnOnIncompatibleVersions, version: packageJsonVersion}
export class XFoo extends LitElement {}
```

I think when the version is packageJsonVersion, the comparison _always_ fails, so that in dev you get the warnings and in prod you get them unless you set up your build system correctly.